app/models/alchemy/json_api/page.rb

```diff
@@ -0,0 +1,27 @@
+module Alchemy
+  module JsonApi
+    class Page < BaseRecord
+      self.table_name = "alchemy_pages"
+
+      include Alchemy::Taggable
+      include Alchemy::Page::PageNatures
+
+      belongs_to :language, class_name: "Alchemy::Language"
+
+      has_many :elements,
+        -> { available.order(:position) },
+        class_name: "Alchemy::JsonApi::Element",
+        inverse_of: :page
+
+      scope :published, -> {
+          where("#{table_name}.public_on <= :time AND " \
+                "(#{table_name}.public_until IS NULL " \
+                "OR #{table_name}.public_until >= :time)", time: Time.current)
+        }
+
+      scope :contentpages, -> { where(layoutpage: false) }
+      scope :layoutpages, -> { where(layoutpage: true) }
+      scope :with_language, ->(language_id) { where(language_id: language_id) }
```

Comment:
We could add a `#nested_elements` method here: 

```rb
def nested_elements
   elements.select { |e| e.parent_element_id == nil } 
end
```

Reply:
Isnt this more `first_level_elements` or `non_nested_elements`?

Reply:
Or reintroduce `all_elements` and use the `elements` relation as the first level elements. So we do not break existing frontend deserializers.